mozilla/addons-server
# link the activity log to the ContentDecision so we can backtrack it

Fixes: mozilla/addons#15228


### Description

Adds an intermediary ActivityLog model that allows us to track ContentDecision that are linked to a reviewer/moderator activity.  

Most of the change here is to enable all actions from Cinder to generate an ActivityLog and link the decision to it.  The change in `notify_reviewer_decision`, that is called from the reviewer tools, is more of an intermediate hack because the activity log was already created - it's going to need more refactoring to create the ContentDecision earlier that will be addressed in other issues.

### Context

Having a link between the `ContentDecision` and the connected activity will unlock more accurate appeal handling (we can flag the exact version that was rejected, rather than guessing at the add-on level); and the 2nd level approvals for reviewer tools actions.  (And potentially drop `CinderJobLog` and/or `CinderPolicyLog` as the `ContentDecision` already links to jobs and policies)

### Testing

Difficult to test functionally because it's a dependency for further work, (on top of the usual fun for testing Cinder integration).

To test Cinder actions:
- Enable all the Cinder/DSA related waffle switches
- Report an add-on (or other content) to Cinder
- Resolve the job in Cinder with a policy that takes down the content
- replay the webhook payload locally
- See the content is deleted/disabled
- Check the latest ActivityLog (it should be the latest, but you may have to dig a little) and see it has a `.contentdecisionlog_set` manager with 1 `ContentDecisionLog` instance, where the `.decision` points to the created CinderDecision.

Bonus testing with appeals:
- appeal the takedown as the content owner (e.g. add-on developer)
- grant the appeal in Cinder (for addons the appeal will be in the T&S Escalations queue)
- replay the webhook payload locally
- see the content is re-enabled
- Check the latest ActivityLog (it should be the latest) and see it has a `.contentdecisionlog_set` manager with 1 `ContentDecisionLog` instance, where the `.decision` points to the created CinderDecision.

### Checklist
